api/routes/auth/certifications/files.test.js

```diff
@@ -0,0 +1,162 @@
+      update: sinon.stub(),
+      digest: sinon.stub(),      
+    },
+    validateDoc: sinon.stub()
+  }
+  
+  endpointTest.beforeEach(() => {
+    app = mockExpress();
+    res = mockResponse();
+    next = sinon.stub();
+  });
+
+  endpointTest.test('setup', async setupTest => {
+    filesEndpoint(app);
+    
+    setupTest.ok(
+      app.post.calledWith(
+        '/auth/certifications/files', 
+        loggedIn,
+        can('edit-state-certifications'), 
+        sinon.match.func,
+        sinon.match.func
+      ),
+      '/auth/certifications/files POST endpoint is setup'
+    );
+
+    setupTest.ok(
+      app.get.calledWith(
+        '/auth/certifications/files/:fileID', 
+        loggedIn,
+        can('view-state-certifications'), 
+        sinon.match.func
+      ),
+      '/auth/certifications/files/:fileId GET endpoint is setup'
+    );
+  });
+  
+  endpointTest.test('POST endpoint for uploading a state certification letter/file', async tests => {
+    let handler; 
+    const buffer = 'buff';    
+    const req = {};
+    
+    tests.beforeEach(async () => {
+      filesEndpoint(app, { ...di });
+      handler = app.post.args[0].pop();
+      di.crypto.createHash.withArgs('sha256').returnsThis();
+      di.crypto.update.withArgs(buffer).returnsThis();
+      di.crypto.digest.withArgs('hex').returns('123');
+    });
+  
+    tests.test('the file is not a valid doc type', async test => {
+      di.validateDoc.resolves({
+        error: 'Unsupported file format'
+      })      
+      req.file = {
+        buffer,
+        size: 1234
+      };      
+      
+      await handler(req, res, next); 
+  
+      test.ok(res.status.calledWith(415), 'sends a 415 error');
+      test.ok(
+        res.send.calledWith({
+          error: 'Unsupported file format'
+        })
+      );
+      test.ok(res.send.calledAfter(res.status), 'response is terminated');
+    });
+    
+    tests.test('with a valid doc the file is saved to storage provider', async test => {
+      di.validateDoc.returns({});     
+      req.file = {
+        buffer,
+        size: 1234
+      };
+      di.putFile.resolves();
+      
+      await handler(req, res, next);
+      
+      test.ok(
+        res.send.calledWith({
+          url: '/auth/certifications/files/123'
+        })
+      );
+    });  
+    
+    tests.test('error persisting file to local or remote storage', async test => {
+      di.validateDoc.returns({});     
+      req.file = {
+        buffer,
+        size: 1234
+      };
+      di.putFile.rejects();
+      
+      await handler(req, res, next);
+
+      test.ok(
+        next.calledWith({ message: 'Unable to upload file' }),
+        'sends error message'
+      );
+    });      
+  });    
+  
+  endpointTest.test('GET endpoint for fetching a state certification letter/file', async tests => {
+    let handler; 
+    
+    tests.beforeEach(async () => {
+      filesEndpoint(app, { ...di });
+      handler = app.get.args[0].pop();
```

Comment:
This should be a named lookup

Reply:
This code was originally created in 3403 and it was updated there before merge so this should be good now.